File: jagraj/19/climbingStairs.py
class Solution:
    def __init__(self):
        self.mem = [0, 1, 2, 3]
        
    def climbStairs(self, n: int) -> int:
        for i in range(4, n+1):
            self.mem.append(self.mem[-1]+self.mem[-2])
        
        return self.mem[n]


if __name__ == '__main__':
    s = Solution()

    for i in range(10):
        print(i+1, s.climbStairs(i+1))

File: jagraj/19/ClimbingStairs.java
public class ClimbingStairs {
    private int[] poop;

    public ClimbingStairs() {
        this.poop = new int[46];
        for (int i = 0; i <= 3; i++) this.poop[i] = i;
    }

    public int climbStairs(int n) {

        for (int i = 4; i <= n; i++) {
            this.poop[i] = this.poop[i-1]+this.poop[i-2];
        }
        return this.poop[n];
    }
}
